packages/expo-calendar/src/Calendar.ts

```diff
@@ -558,6 +558,149 @@ export type DaysOfTheWeek = {
 
 export { PermissionResponse, PermissionStatus, PermissionHookOptions };
 
+/**
+ * The result of presenting the calendar dialog.
+ * These constants map to [`EKEventEditViewAction`](https://developer.apple.com/documentation/eventkitui/ekeventeditviewaction) and [`EKEventViewAction`](https://developer.apple.com/documentation/eventkitui/ekeventviewaction).
+ * @platform ios
+ * */
+export type CalendarDialogResultActionsIos = 'done' | 'canceled' | 'deleted' | 'responded';
+
+/**
+ * The result of launching the calendar Activity.
+ * Android doesn't provide enough information to determine the user's action, so the result is always 'done'.
+ * The user may have cancelled the dialog, saved the event, or deleted it.
+ * @platform android
+ * */
+export type CalendarDialogResultActionsAndroid = 'done';
+
+/**
+ * The result of presenting a calendar dialog. It indicates what action the user took (e.g. canceled the dialog).
+ * The information made available by Android is less complete compared to iOS.
+ * @header systemProvidedUI
+ * */
+export type CalendarDialogResult = {
+  action: CalendarDialogResultActionsIos | CalendarDialogResultActionsAndroid;
+};
+
+/**
+ * The result of creating an event in the calendar dialog on iOS.
+ * @platform ios
+ * */
+export type CalendarDialogCreatedEventResultIos = {
+  action: 'saved';
```

Comment:
Can we change it to 'done' or change the Android type to 'saved'? It's weird that after creating an event users would have to handle multiple values even if they're meaning the same.

Reply:
on iOS, it's easy to tell that an event was saved, and get its ID.

on Android, we don't really know if the user saved an event, cancelled the dialog, or etc - the intent result doesn't say that. That's why 'done' is always returned on Android.

I agree it's inconvenient but I'm afraid users will have to handle the results themselves, depending on what their needs are.